qemu/tests/qemu_guest_agent.py

```diff
@@ -1375,6 +1373,134 @@ def setup(self, test, params, env):
+        file_name = "/tmp/fsfreeze_%s.txt" % flag
+        process.system("rm -rf %s" % file_name)
+        self.vm.copy_files_from("C:\\fsfreeze.txt", file_name)
+        with open(file_name, 'r') as f:
+            list_time = f.readlines()
+
+        for i in list(range(0, len(list_time))):
+            list_time[i] = list_time[i].strip('')
+
+        for i in list(range(1, len(list_time))):
+            num_d = float(list_time[i]) - float(list_time[i - 1])
+            if num_d > 8:
+                logging.info("Time stamp is not continuous,"
+                             " so the FS is frozen.")
+                fs_status = "frozen"
+                break
+        if not fs_status == flag:
+            self.test.fail("FS is not %s, it's %s." % (flag, fs_status))
+
+    @error_context.context_aware
+    def _action_before(self):
+        """
+        Before freeze or thaw guest file system, start a background test.
+        """
+        session = self._get_session(self.params, None)
+        self.session = session
+        self._open_session_list.append(session)
+
+        error_context.context("Before freeze/thaw the FS, run the background "
+                              "job.", logging.info)
+        logging.info("Write time stamp to guest file per second "
+                     "as a background job.")
+        fswrite_cmd = utils_misc.set_winutils_letter(
+            self.session, self.params["gagent_fs_test_cmd"])
+
+        self.fswrite_thread = utils_misc.InterruptedThread(self.session.cmd,
+                                                           args=(fswrite_cmd,
+                                                                 120))
+        self.fswrite_thread.start()
+
+    @error_context.context_aware
+    def _action_after(self, flag, write_timeout):
+        """
+        After freeze or thaw guest file system,check if it works.
+
+        :param flag: check flag for frozen or thaw
+        :param write_timeout: timeout for writing to guest file
+
+        """
+        if self.fswrite_thread:
+            error_context.context("Waiting %s, then finish writing the time "
+                                  "stamp in guest file." % write_timeout)
+            self.fswrite_thread.join(write_timeout)
+            if self.fswrite_thread.isAlive():
+                session2 = self._get_session(self.params, None)
+                self._open_session_list.append(session2)
+                k_cmd = "wmic process where \"name='python.exe' and " \
+                        "CommandLine Like '%fsfreeze%'\" call terminate"
+                s, o = session2.cmd_status_output(k_cmd)
+                if s:
+                    self.test.error("Command '%s' failed, status: %s, "
+                                    "output: %s" % (k_cmd, s, o))
+        error_context.context("Check guest FS status.", logging.info)
+        self._fswrite_check(flag)
+
+    @error_context.context_aware
+    def gagent_check_fsfreeze(self, test, params, env):
```

Comment:
Is your base the latest? There is function `gagent_check_fsfreeze` here: https://github.com/autotest/tp-qemu/blob/master/qemu/tests/qemu_guest_agent.py#L1190 Why you have it newly added in the patch?

Reply:
Sorry, I saw you override it for subclass `QemuGuestAgentBasicCheckWin`.